app/plugins/webview-preload.js

```diff
@@ -28,7 +27,8 @@ if (micromatch.isMatch(window.location.origin, originsAllowed) || protocolsAllow
       });
     });
     setTimeout(() => {
-      ipcRenderer.sendTo(workerWebContentsId, 'bx-api-perform', channel, payload);
+      ipcRenderer.invoke('get-worker-contents-id')
```

Comment:
Perhaps store the result of this call so we just have to call it once per process.

Reply:
> Perhaps store the result of this call so we just have to call it once per process.

I have doubts about this suggestion
Should we create a module for this purpose?
and then use it like 

```javascript
  const id = require('some-new-module').workerWebContentsId
```

What is the right place to put it?
Is this `worker-id` available in any moment of time and code? What if we ask for it too early?

Reply:
A module seems a little overkill, I was more thinking just storing the result of the call in a variable in the file itself

Reply:
That was my first version :)
Like this

```javascript
  const workerWebContentsIdPromise = ipcRenderer.invoke('get-worker-contents-id');

  const sendPerformToProxy = (channel, payload) => {
    const p = new Promise(resolve => {
      ipcRenderer.once(`bx-api-perform-response-${channel}`, (_, result) => {
        resolve(result);
      });
    });

    setTimeout(() => {
        workerWebContentsIdPromise
            .then(workerWebContentsId => ipcRenderer.sendTo(workerWebContentsId, 'bx-api-perform', channel, payload));
    }, 1);
    return p;
  };
```

But the same concern about requesting it too early.
And `setTimeout` here looks like brute workaround, so I decided not to touch it and to keep as is.